clustered/common/src/test/java/org/ehcache/clustered/common/ServerStoreCompatibilityTest.java

```diff
@@ -0,0 +1,485 @@
+                                                                                ACTUAL_VALUE_TYPE,
+                                                                                KEY_SERIALIZER_TYPE,
+                                                                                VALUE_SERIALIZER_TYPE,
+                                                                                Consistency.STRONG);
+
+    ServerStoreCompatibility serverStoreCompatibility = new ServerStoreCompatibility();
+
+    try
+    {
+      serverStoreCompatibility.verify(serverConfiguration, clientConfiguration);
+      fail("Expected ClusteredStoreValidationException");
+    }
+    catch(ClusteredStoreValidationException e)
+    {
+      Assert.assertTrue("test failed", e.getMessage().equals(ERROR_MESSAGE_BASE + "resourcePoolSharedPoolName existing: " + ((Shared)serverConfiguration.getPoolAllocation()).getResourcePoolName() + ", desired: " + ((Shared)clientConfiguration.getPoolAllocation()).getResourcePoolName()));
+    }
+  }
+
+  @Test
+  public void testAllResourceParametersMatch()
+  {
+    ServerStoreConfiguration serverConfiguration = new ServerStoreConfiguration(FIXED_POOL_ALLOCATION,
+                                                                                STORED_KEY_TYPE,
+                                                                                STORED_VALUE_TYPE,
+                                                                                ACTUAL_KEY_TYPE,
+                                                                                ACTUAL_VALUE_TYPE,
+                                                                                KEY_SERIALIZER_TYPE,
+                                                                                VALUE_SERIALIZER_TYPE,
+                                                                                Consistency.EVENTUAL);
+
+    ServerStoreConfiguration clientConfiguration = new ServerStoreConfiguration(FIXED_POOL_ALLOCATION,
+                                                                                STORED_KEY_TYPE,
+                                                                                STORED_VALUE_TYPE,
+                                                                                ACTUAL_KEY_TYPE,
+                                                                                ACTUAL_VALUE_TYPE,
+                                                                                KEY_SERIALIZER_TYPE,
+                                                                                VALUE_SERIALIZER_TYPE,
+                                                                                Consistency.EVENTUAL);
+
+    ServerStoreCompatibility serverStoreCompatibility = new ServerStoreCompatibility();
+
+    try
+    {
+      serverStoreCompatibility.verify(serverConfiguration, clientConfiguration);
+    }
+    catch(Exception e)
+    {
+      fail("No Exception Expected!");
```

Comment:
It's much better to declare the test method with `throws Exception` than catching it and failing the test without knowing what the exception was.